lib/ansible_tower_client/collection_methods.rb

```diff
@@ -0,0 +1,32 @@
+module AnsibleTowerClient
+  module CollectionMethods
+    def all
+      body = JSON.parse(Api.get(mapping).body)
+      results = body["results"]
+      loop do
+        break if body["next"].nil?
+        body = JSON.parse(Api.get(body["next"]).body)
+        results += body["results"]
+      end
+
+      results.collect { |raw| new(raw) }
+    end
+
+    def find(id)
+      body = JSON.parse(Api.get("#{mapping}/#{id}/").body)
+      raise NoMethodError if body['id'].nil?
```

Comment:
I realize this should probably be a custom or Faraday based exception - I have NoMethodError error there now mostly as a placeholder. Wouldn't mind thoughts on this.